astroplan/constraints.py

```diff
@@ -0,0 +1,565 @@
+            Target or list of targets
+
+        observer : `~astroplan.Observer`
+            The observer who has constraints ``constraints``
+
+        time_resolution : `~astropy.units.Quantity` (optional)
+            Set the time resolution in calculations of the altitude/azimuth
+
+        Returns
+        -------
+        altaz_dict : dict
+            Dictionary containing two key-value pairs. (1) 'times' contains the
+            times for the alt/az computations, (2) 'altaz' contains the
+            corresponding alt/az coordinates at those times.
+        """
+        if not hasattr(observer, '_altaz_cache'):
+            observer._altaz_cache = {}
+
+        times = time_grid_from_range(time_range,
+                                     time_resolution=time_resolution)
+
+        # convert times, targets to tuple for hashing
+        aakey = (tuple(times.jd), tuple(targets))
+
+        if aakey not in observer._altaz_cache:
+            if force_zero_pressure:
+                observer_old_pressure = observer.pressure
+                observer.pressure = 0
+
+            altaz = observer.altaz(times, targets)
+            observer._altaz_cache[aakey] = dict(times=times,
+                                                altaz=altaz)
+            if force_zero_pressure:
+                observer.pressure = observer_old_pressure
+
+        return observer._altaz_cache[aakey]
+
+class AltitudeConstraint(Constraint):
+    """
+    Constrain the altitude of the target.
+
+    Note: this will misbehave if you try to constrain negative altitudes, as
+    the `~astropy.coordinates.AltAz` frame tends to mishandle negative
+    altitudes.
+    """
+    def __init__(self, min=None, max=None):
+        """
+        Parameters
+        ----------
+        min : `~astropy.units.Quantity` or `None`
+            Minimum altitude of the target. `None` indicates no limit.
+
+        max : `~astropy.units.Quantity` or `None`
+            Maximum altitude of the target. `None` indicates no limit.
+        """
+        if min is None:
+            self.min = 0*u.deg
+        else:
+            self.min = min
+        if max is None:
+            self.max = 90*u.deg
+        else:
+            self.max = max
+
+    def _compute_constraint(self, time_range, observer, targets,
+                            time_resolution=DEFAULT_TIME_RESOLUTION):
+
+        cached_altaz = self._get_altaz(time_range, observer, targets,
+                                       time_resolution=time_resolution)
+        altaz = cached_altaz['altaz']
+        lowermask = self.min < altaz.alt
+        uppermask = altaz.alt < self.max
+        return lowermask & uppermask
+
+class AirmassConstraint(AltitudeConstraint):
+    """
+    Constrain the airmass of a target.
+
+    The airmass is approximated by the secant of the zenith angle.
+    """
+    def __init__(self, max=None, min=None):
+        """
+        Note: the ``max`` and ``min`` arguments appear in the order (max, min)
```

Comment:
Definitely good to include this note (although see my comment above about using ``.. note::`` to make it slightly prettier).  But I suggest just being even a bit more explicit and including an example along the lines of 
```
Examples
--------
To create a constraint that requires the airmass be better than 2::

    AirmassConstraint(2)

```
I think just about anyone would read that and understand it to be "the airmass value should be less than 2, meaning closer to zenith, not closer to horizon... Whereas the current text is still subject to confusion by the in-a-hurry observer who thinks of "min" as meaning "lowest altitude" because altitude and airmass are sort of mixed together in their head (hint: a good example of such a person is me ;)